Tests/Functional/SelectedTest/main.swift

```diff
@@ -0,0 +1,58 @@
+// RUN: %{swiftc} %s -o %{built_tests_dir}/SelectedTest
+// RUN: %{built_tests_dir}/SelectedTest SelectedTest.ExecutedTestCase/test_foo > %t || true
+// RUN: %{built_tests_dir}/SelectedTest SelectedTest.ExecutedTestCase >> %t || true
+// RUN: %{built_tests_dir}/SelectedTest >> %t || true
+// RUN: %{xctest_checker} %t %s
+//// Output from the first invocation (running a single test method)
+// CHECK: Test Case 'ExecutedTestCase.test_foo' started.
+// CHECK: Test Case 'ExecutedTestCase.test_foo' passed \(\d+\.\d+ seconds\).
+// CHECK: Executed 1 test, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+// CHECK: Total executed 1 test, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+//// Output from the second invocation (running one entire test case)
+// CHECK: Test Case 'ExecutedTestCase.test_bar' started.
+// CHECK: Test Case 'ExecutedTestCase.test_bar' passed \(\d+\.\d+ seconds\).
+// CHECK: Test Case 'ExecutedTestCase.test_foo' started.
+// CHECK: Test Case 'ExecutedTestCase.test_foo' passed \(\d+\.\d+ seconds\).
+// CHECK: Executed 2 tests, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+// CHECK: Total executed 2 tests, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+//// Output from the third invocation (running all tests)
+// CHECK: Test Case 'ExecutedTestCase.test_bar' started.
+// CHECK: Test Case 'ExecutedTestCase.test_bar' passed \(\d+\.\d+ seconds\).
+// CHECK: Test Case 'ExecutedTestCase.test_foo' started.
+// CHECK: Test Case 'ExecutedTestCase.test_foo' passed \(\d+\.\d+ seconds\).
+// CHECK: Executed 2 tests, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+// CHECK: Test Case 'SkippedTestCase.test_baz' started.
+// CHECK: Test Case 'SkippedTestCase.test_baz' passed \(\d+\.\d+ seconds\).
+// CHECK: Executed 1 test, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+// CHECK: Total executed 3 tests, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
```

Comment:
For this test in particular, I don't think it's going to work to reorganize the `CHECK` lines, unfortunately, because of the way that this one executes the tests repeatedly in order to compare the output when running them with different test filters active.

Or is there a neat way to structure this that I haven't thought of yet?